components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java

```diff
@@ -46,6 +48,13 @@
 
     private UserAdminStub userAdminStub;
 
+    // A static variable is used to track if pagination support from the remote API is verified, as the client is
+    // initiated per request.
+    private static boolean paginationSupportVerified = false;
```

Comment:
why do we need to variables, wouldn't storing whether 'paginationSupported' not enough?

Reply:
Yes, we need this to track if pagination support from the remote API is verified, as the client is initiated per request. We used the same logic in the SP pagination also,

https://github.com/wso2/carbon-identity-framework/blob/91e9a22ea02d9d3f197c7d66cae96bffcdce9f94/components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/client/ApplicationManagementServiceClient.java#L55

Reply:
This is the master branch. No need to have a config to check paginationSupported or not. It is supported OTB.